doc/design/audit-trail-mvp.md

```diff
@@ -0,0 +1,156 @@
+what that means. We are choosing to fulfill the requirement with a feature that
+indicates to Dandiset owners and admins the sequence of significant actions
+taken to modify Dandisets, along with identification of who performed each
+action.
+
+The major motivation is to inform owners about how their Dandisets have changed
+when a given owner did not make the change and wishes to know who to speak to
+about those changes. In essence, it is a tool to support effective
+collaboration.
+
+## Requirements
+
+1. **Persistent tracking of significant actions.** Whenever a *significant
+   action* (see list [below](#significant-actions)) occurs successfully, the archive will generate a
+   persistent record of that action, including the following data:
+    - The date and time
+    - The authorized user who triggered the action (note that this may be an actual user or the system itself for internal automated actions)
+    - The ID of the Dandiset on which the action occurs
+    - Event-specific data about the action itself
+
+2. **Display of audit log (to owners and admins only).** The Dandiset landing
+   page will display a rendered (i.e. human-readable) version of the log event
+   data (as long as the viewing user is an owner of the Dandiset, or an admin).
+
+   A possible example of such a display follows (the final rendered form may
+   differ from this depending on best practices, etc.):
+    
+    ```
+    YYYYMMDDTHHMMSS.mmmm: X added asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X updated asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X removed asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X updated dandiset metadata
+    ```
+    
+3. **Extensible significant action list.** The significant actions the system is
+   capable of tracking should be extensible--that is, in the future new actions
+   should be able to be added to the system, and existing actions should be able
+   to expand the specific data they carry.
+
+## Significant Actions
+
+The following is the list of actions performed on Dandisets that are considered
+"significant" (i.e., require an audit record to be generated), along with any
+additional, event-specific data each will carry with it.
+
+|  | General data | Specific data |
+| --- | --- | --- |
+| Create Dandiset | date_time, user_id, dandiset_id | metadata |
+| Add owner | date_time, user_id, dandiset_id | owner_id |
+| Remove owner | date_time, user_id, dandiset_id | owner_id |
+| Update Dandiset metadata | date_time, user_id, dandiset_id | metadata |
+| Add asset | date_time, user_id, dandiset_id | asset_path, checksum, metadata, blob_id |
+| Update asset | date_time, user_id, dandiset_id | asset_path, checksum, blob_id, metadata |
+| Remove asset | date_time, user_id, dandiset_id | asset_path |
+| Publish Dandiset | date_time, user_id, dandiset_id | version |
+| Delete Dandiset | date_time, user_id, dandiset_id | none |
```

Comment:
@jwodder do you think the list of actions above, and data for them would be sufficient for `backups2datalad` to perform synchronization to at least the level it does now, or some data would be missing?

Reply:
`backups2datalad` also needs exact information on:

* Creation, replacement/modification, and deletion of individual entries within a Zarr
* Unembargoing of Dandisets

Also, although `backups2datalad` doesn't currently use this, I think deletions of Dandiset versions should be recorded as well.

Reply:
We do need to add embargo/unembargo events to the list so those operations can be part of the event log.

Though I said in a different thread that I don't think we should make a special case for Zarr operations, perhaps I am wrong (since, as you're pointing out, there are indeed special ways in which Zarrs are manipulated). Still, it doesn't seem quite right, and I want to be careful not to overspecify the audit feature to work precisely with `backups2datalad`, lest we lose sight of "audit", which is the real goal here.

If this is important, I think we should discuss it in a meeting.

Reply:
zarr: relates to the disclaimer I suggested to add that we do not want in this MVP to cover blob/zarr level information https://github.com/dandi/dandi-archive/pull/1801/files#r1467945249 . Without such explicit "scoping" statement MVP then IMHO should include blob/zarr level basic audit

unembargo: TL;DR -- should we actually remove "Unembargo Dandiset" and rely on changes to metadata?

Rationale:   given that we doing "asset level" audit and unembargoing is to happen (in [redesign](https://github.com/dandi/dandi-archive/pull/1802/files#diff-f3d72d358571f346f0a3e26e3bf309127d6af6aac1e39f60e291b947d86eeb19R132)) on assets level, it feels like we might want to add "Unembargo asset".  But "embargoed" is also a state within metadata, and we already have "Update asset" with metadata being recorded, so at least we will have information collected.

But then should we make it symmetric at Dandiset level where ATM we have two "Update Version metadata" and "Unembargo Dandiset". Should we remove "Unembargo Dandiset" and rely on discoverying that action through changes to its metadata?

Reply:
We will likely want to record that an asset went from hidden to open, etc., but since that's not a capability in the system yet, we can wait until that lands to design its appropriate audit events.

I think keeping "unembargo dandiset" as its own significant action is important. We do indeed record that state in the metadata, but that doesn't mean the event needs to be swallowed up in the "update metadata" event -- it is special enough to warrant the event.